### PR TITLE
LabelSelector should not be added while restoring pvc

### PR DESCRIPTION
**Pull request by muxuelanKK**
When old pvc had pv selector, new pvc should not include it. 
image

### CODE REVIEW

**Review by akalenyu — COMMENTED** [review]
Thanks for the PR!

I think the labelselector is just one symptom of a broader issue where we just copy out the entire spec of the source PVC.
Maybe we should take look at only setting fields we care about instead of https://github.com/kubevirt/kubevirt/blob/817c3adf0adf966d625c4c0efa4ebb6c55e47219/pkg/storage/snapshot/restore.go#L998

**Comment by mhenriks** [comment]
@muxuelanKK is this affecting the restore at all?  Curious if/how PV label selectors could have any effect when creating a PVC with VolumeSnapshot datasource

@akalenyu I think this is the only field that could potentially be problematic. What do you think?

**Comment by akalenyu** [comment]
> @muxuelanKK is this affecting the restore at all? Curious if/how PV label selectors could have any effect when creating a PVC with VolumeSnapshot datasource
> 
> @akalenyu I think this is the only field that could potentially be problematic. What do you think?

This https://github.com/kubevirt/kubevirt/blob/817c3adf0adf966d625c4c0efa4ebb6c55e47219/pkg/storage/snapshot/restore.go#L1041 is a bit clunky too, alongside datasource&datasourceref. Maybe we could just set them all explicitly, that would be nicer IMO

**Comment by muxuelanKK** [comment]
> > @muxuelanKK is this affecting the restore at all? Curious if/how PV label selectors could have any effect when creating a PVC with VolumeSnapshot datasource
> > @akalenyu I think this is the only field that could potentially be problematic. What do you think?
> 
> This
> 
> https://github.com/kubevirt/kubevirt/blob/817c3adf0adf966d625c4c0efa4ebb6c55e47219/pkg/storage/snapshot/restore.go#L1041
> 
> is a bit clunky too, alongside datasource&datasourceref.
> Maybe we could just set them all explicitly without copying, that would be nicer IMO

Yes, hold on

**Comment by muxuelanKK** [comment]
> > > @muxuelanKK is this affecting the restore at all? Curious if/how PV label selectors could have any effect when creating a PVC with VolumeSnapshot datasource
> > > @akalenyu I think this is the only field that could potentially be problematic. What do you think?
> > 
> > 
> > This
> > https://github.com/kubevirt/kubevirt/blob/817c3adf0adf966d625c4c0efa4ebb6c55e47219/pkg/storage/snapshot/restore.go#L1041
> > 
> > is a bit clunky too, alongside datasource&datasourceref.
> > Maybe we could just set them all explicitly without copying, that would be nicer IMO
> 
> Yes, hold on

done

**Review by akalenyu — APPROVED** [review]
one note - may want to add a test in `pkg/storage/snapshot/restore_test.go` to make sure we don't regress over this in the future

File: pkg/storage/snapshot/restore.go [hunk 1]
```diff
@@ -995,7 +995,12 @@ func CreateRestorePVCDef(restorePVCName string, volumeSnapshot *vsv1.VolumeSnaps
 			Labels:      sourcePVC.Labels,
 			Annotations: sourcePVC.Annotations,
 		},
-		Spec: sourcePVC.Spec,
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes:      sourcePVC.Spec.AccessModes,
+			Resources:        sourcePVC.Spec.Resources,
+			StorageClassName: sourcePVC.Spec.StorageClassName,
+			VolumeMode:       sourcePVC.Spec.VolumeMode,
+		},
```
Comment (mhenriks) [inline pkg/storage/snapshot/restore.go:1003]: Given this new pattern of explicitly setting the fields we care about I think it makes sense to set dataSource and dataSourceRef in in this block as well

Comment (mhenriks) [reply pkg/storage/snapshot/restore.go:1003]: @muxuelanKK you planning to come back to this?

**Pull request merged**